lib/MusicBrainz/Server/Edit/Artist/Edit.pm

```diff
@@ -193,9 +193,39 @@ around allow_auto_edit => sub {
     return 0 if exists $self->data->{old}{end_area_id} &&
         defined($self->data->{old}{end_area_id}) && $self->data->{old}{end_area_id} != 0;
 
-    return 0 if $self->data->{new}{ipi_codes};
+    if (defined $self->data->{new}{ipi_codes}) {
+        # If there's already IPIs for the artist, not an autoedit
+        if (defined $self->data->{old}{ipi_codes} &&
+            scalar @{$self->data->{old}{ipi_codes}} != 0) {
+                return 0;
+        }
+
+        # If there's already an entity with any of the IPIs, not an autoedit
+        my $reused_ipis = $self->c->model('Artist')->find_reused_ipis(
+            $self->data->{new}{ipi_codes}
+        );
 
-    return 0 if $self->data->{new}{isni_codes};
+        if (%$reused_ipis) {
+            return 0;
+        }
+    }
+        
+    if (defined $self->data->{new}{isni_codes}) {
+        # If there's already ISNIs for the artist, not an autoedit
+        if (defined $self->data->{old}{isni_codes} &&
+            scalar @{$self->data->{old}{isni_codes}} != 0) {
+                return 0;
+        }
+
+        # If there's already an entity with any of the ISNIs, not an autoedit
+        my $reused_isnis = $self->c->model('Artist')->find_reused_isnis(
+            $self->data->{new}{isni_codes}
+        );
+
+        if (%$reused_isnis) {
+            return 0;
+        }
```

Comment:
Something is wrong with this test. I tested entering an existing ISNI for another artist. Even though the ModBot added a note, the edit still passed through as an auto-edit. It seems worth adding to `t/`.

Reply:
Did you add it as an autoeditor?

Reply:
Probably but I dropped the test DB since then. I just tested again and it isn’t applied when not having autoeditor privs.